packages/solid/src/components/avatar/primitives.tsx

```diff
@@ -16,30 +17,38 @@ import { splitProps } from 'solid-js'
 /**
  * The root component of the Avatar.
  */
-export function AvatarRoot(props: AvatarRootProps & AvatarVariantProps) {
-  const [{ gradient, shape, size }, rootProps] = splitProps(props, [
-    'gradient',
-    'shape',
-    'size',
-  ])
+export function AvatarRoot(
+  props: AvatarRootProps & AvatarVariantProps & WithCss,
+) {
+  const [{ gradient, shape, size, css: customCss }, rootProps] = splitProps(
+    props,
+    ['gradient', 'shape', 'size', 'css'],
+  )
   const styles = avatar({ gradient, shape, size })
-  return <Avatar.Root {...rootProps} class={cx(styles.root, rootProps.class)} />
+  return <Avatar.Root {...rootProps} class={cx(styles.root, css(customCss))} />
```

Comment:
I think you want to do something like this

```
const styles = avatar.raw({gradient, shape, size}); // notice raw this returns an object
return <Avatar.Root {...rootProps} class={cx(css(styles.root, customCss), rootProps.class} /> 
```
Use the `css` function to merge the recipe **and** the provided css props.
Use the `cx` function to join the class names produced by the `css` function  **and** the `class` provided by the consumer.

Reply:
@cgatian I appreciate that insight. Cerberus recipes are config not `s/cva` which allow us to utilize this pattern (since they are compiled differently). 

[Check out the section](https://panda-css.com/docs/concepts/merging-styles#merging-config-recipe-and-style-object)

Am I overlooking something there?

Reply:
Oh OK yeah I didnt know what `avatar` type was. I think you're right. Only thought might be to make sure that `rootProps.class` gets merged into `class` but I don't know anything about Solid

Reply:
Yeah, all of our recipes are config ones. I did think about that but left if out to prevent initial confusion about how you could customize - but in hindsight I think it's definitely a good idea to keep it in with the `css` prop.

Apparently co-pilot does too. Lol

Reply:
Done